libsolidity/ast/Types.cpp

```diff
@@ -2146,30 +2147,79 @@ TypePointer StructType::interfaceType(bool _inLibrary) const
 
 bool StructType::canBeUsedExternally(bool _inLibrary) const
 {
-	if (_inLibrary && location() == DataLocation::Storage)
```

Comment:
Yeah, I think whether or not the location is storage is not considered anymore.

Reply:
It wasn't always considered before either though, e.g. if the location is memory, _inlibrary is true but all the types can be used externally in the struct. How should the new behavior be?

Reply:
The old behaviour was: If the location is not storage, `_inLibrary` is ignored.
The new behaviour:
 - disallow internal function pointers regardless of storage location or library-flag
 - otherwise identical with old behaviour, i.e.:

- allow recursive structs if location is storage and library-flag is set.
- allow mapping members if location is storage and library-flag is set